log4j2-ecs-layout/src/test/java/co/elastic/logging/log4j2/EcsLayoutWithStackTraceAsArrayTest.java

```diff
@@ -50,7 +50,7 @@ void testLogException() throws Exception {
         assertThat(log.get("error.stack_trace").isArray()).isTrue();
         ArrayNode arrayNode = (ArrayNode) log.get("error.stack_trace");
         assertThat(arrayNode.size()).isEqualTo(4);
-        assertThat(arrayNode.get(0).textValue()).isEqualTo("java.lang.NumberFormatException: For input string: \"NOT_AN_INT\"");
+        assertThat(arrayNode.get(0).textValue()).isEqualTo("java.lang.NumberFormatException: For input string: \"NOT_AN_INT\"\r");
```

Comment:
Looks platform-specific. Does this fail for you without that? 
If so, better to change to use `startsWith` or regex comparison instead

Reply:
Test on CI (Ubuntu 18) fails with:
```
[2022-05-01T12:33:28.698Z] Failed tests: 
[2022-05-01T12:33:28.698Z]   EcsLayoutWithStackTraceAsArrayTest.testLogException:53 
[2022-05-01T12:33:28.698Z] Expecting:
[2022-05-01T12:33:28.698Z]  <"java.lang.NumberFormatException: For input string: "NOT_AN_INT"">
[2022-05-01T12:33:28.698Z] to be equal to:
[2022-05-01T12:33:28.698Z]  <"java.lang.NumberFormatException: For input string: "NOT_AN_INT"
">
[2022-05-01T12:33:28.698Z] but was not.
```

Reply:
Yes, it fails on Windows otherwise. Added a trim before, should be platform-independent then.

Reply:
I don't think this is an issue in the tests but a bug here:

https://github.com/elastic/ecs-logging-java/blob/b09574fe2bf6e4af6cde7b2c7ac2c86a9f43ee49/ecs-logging-core/src/main/java/co/elastic/logging/EcsJsonSerializer.java#L39

It always splits on `\n` instead of depending on the platform.

Reply:
Well yeah this is potentially a bug, agree. Should I change it?